### PR TITLE
Python module

### PR DESCRIPTION
**Pull request by rasapala**

### CODE REVIEW

File: src/pythoninterpretermodule.hpp [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+#include "module.hpp"
+
+namespace py = pybind11;
+
+namespace ovms {
+class Server;
+
+class PythonInterpreterModule : public Module {
+public:
+    PythonInterpreterModule(ovms::Server& ovmsServer);
```
Comment (atobiszei) [inline src/pythoninterpretermodule.hpp:30]: Why do you need server in constructor?

File: src/pythoninterpretermodule.hpp [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#pragma once
+#include <memory>
+
+#include <pybind11/embed.h>  // everything needed for embedding
```
Comment (atobiszei) [inline src/pythoninterpretermodule.hpp:19]: Why include this in header when we use it only in cpp file?

File: src/server.cpp [hunk 3]
```diff
@@ -55,6 +55,8 @@
 
 #if (PYTHON_DISABLE == 0)
 #include <pybind11/embed.h>
+
+#include "pythoninterpretermodule.hpp"
```
Comment (atobiszei) [inline src/server.cpp:59]: L60-61 shouldn't be included here now at all.

File: src/server.cpp [hunk 4]
```diff
@@ -239,6 +246,11 @@ Status Server::startModules(ovms::Config& config) {
     // GRPC & REST
     Status status;
     bool inserted = false;
+#if (PYTHON_DISABLE == 0)
+    auto itPython = modules.end();
```
Comment (atobiszei) [inline src/server.cpp:250]: move:
auto it = modules.end() above ifdef and dont introduce another iterator since we don't use it later anyway.

**Review by mzegla — APPROVED** [review]

**Review by atobiszei — APPROVED** [review]

**Pull request merged**